CoreGraphUtilities/src/au/gov/asd/tac/constellation/graph/utilities/hashmod/Hashmod.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2010-2020 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.utilities.hashmod;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import static javatests.TestSupport.fail;
+
+/**
+ * A text hashmod based on a supplied CSV file.  Will modify attributes
+ * specified in the headers to be values based on the first Key column.
+ *
+ * @author CrucisGamma
+ */
+public class Hashmod {
+
+    public static final String ATTRIBUTE_NAME = "hashmod";
+    private HashmodCSVImportFileParser parser;
+    private String csvFileStr;
+    private List<String[]> data;
+
+    public Hashmod() {
+        parser = null;
+        csvFileStr = "";
+    }
+
+    /**
+     * Create a new Hashmod.
+     *
+     * @param csvFile Name of the CSV file the user has chosen
+     */
+    public Hashmod(final String csvFileStr) {
+        if (csvFileStr == null) {
+            this.csvFileStr = "";
+        } else {
+            this.csvFileStr = csvFileStr;
+        }
+
+        parser = new HashmodCSVImportFileParser();
+        try {
+            data = parser.parse(new HashmodInputSource(new File(csvFileStr)), null);
+        } catch (IOException ex) {
+            fail("IO Exception : " + ex.getLocalizedMessage());
```

Comment:
Maybe change this to a log entry instead of using the fail function (which from what I can tell is made for unit tests)